pkg/storage/simple/manager.go

```diff
@@ -286,10 +286,11 @@ func (c *Chart) Delete(ctx context.Context, version string) error {
 		return ErrorLocking.Format("version", c.Space.Name()+"/"+c.Name()+"/"+version)
 	}
 	err := deleteKeys(ctx, c.Space.SpaceManager.Backend, path.Join(c.Prefix, version), true)
+	// unlock before return
+	lock.Unlock()
```

Comment:
No. It calls function `Delete` which has an write lock too. So it should be release immediately.